tests/unit/test_docs.py

```diff
@@ -28,3 +31,77 @@ def test_generate_docs():
 
     names = {contract["name"] for contract in data["contracts"]}
     assert "PatientDemographics" in names
+
+    # Find all series strings
+    all_series = [
+        paragraph["series"]
+        for spec in data["specs"]
+        for section in spec["sections"]
+        for paragraph in section["paragraphs"]
+    ]
+    # assert that a series string has no leading whitespace for the first and last lines, and
+    # other lines have a multiple of 4 spaces
+    for series in all_series:
+        lines = series.split("\n")
+        for i, line in enumerate(lines):
+            if i in [0, len(lines) - 1]:
+                assert len(line.strip()) == len(line)
+            else:
+                leading_whitespace_count = len(line) - len(line.strip())
+                assert leading_whitespace_count > 0
+                assert leading_whitespace_count % 4 == 0
+
+
+@pytest.mark.parametrize(
+    "source_lines,source_index,expected",
+    [
+        (
+            # single line
+            ['p.d1.is_before("2000-01-20"),'],
+            0,
+            'p.d1.is_before("2000-01-20")',
+        ),
+        (
+            # multiple lines
+            [
+                "case(",
+                "    when(p.i1 < 8).then(p.i1),",
+                "    when(p.i1 > 8).then(100),",
+                "),",
+            ],
+            0,
+            "case(\n    when(p.i1 < 8).then(p.i1),\n    when(p.i1 > 8).then(100),\n)",
+        ),
+        (
+            # real test function; multiple lines, series starts after table_data
+            [
+                "    spec_test(",
+                "        table_data,",
+                "        case(",
+                "            when(p.i1 < 8).then(p.i1),",
+                "            when(p.i1 > 8).then(100),",
+                "        ),",
+                "        {",
+                "            1: 6,",
+                "            2: 7,",
+                "            3: None,",
+                "            4: 100,",
+                "            5: None,",
+                "        },",
+                "    )",
+            ],
+            2,
+            "case(\n    when(p.i1 < 8).then(p.i1),\n    when(p.i1 > 8).then(100),\n)",
+        ),
+        (
+            # incomplete series definition; this should never happen as it should only exist if
+            # there's a syntax error in a specs tests, which would raise an error earlier than
+            # this code
+            ["p.d1.is_before("],
+            0,
+            "p.d1.is_before(",
+        ),
+    ],
+)
+def test_get_series_code(source_lines, source_index, expected):
```

Comment:
I only just noticed that there is a `test_specs.py` which this test belongs in (as this is testing just a bit of `specs.py`, not the docs generation)